server/src/main/java/com/cloud/ha/HighAvailabilityManagerImpl.java

```diff
@@ -294,6 +295,10 @@ public void scheduleRestartForVmsOnHost(final HostVO host, boolean investigate)
                 "Host [" + hostDesc + "] is down." + ((sb != null) ? sb.toString() : ""));
 
         for (VMInstanceVO vm : reorderedVMList) {
+            VMInstanceVO finalVm = vm;
+            if (pendingHaWorks.stream().anyMatch(h -> h.getInstanceId() == finalVm.getId())) {
+                continue;
```

Comment:
@shwstppr , should we/does it make sense to add a debug or trace statement here?